status-page.py

```diff
@@ -1,72 +1,133 @@
 import json
 import sys
+from urllib.parse import urlparse
+
 import requests
 
 
 class StatusPage:
     def __init__(self):
         config_data_file = open(sys.argv[1])
-        self.test_results = open(sys.argv[2])
+        log_file = open(sys.argv[2])
         self.config_json = json.load(config_data_file)
+        self.log_json = json.load(log_file)
         self.access_token = self.config_json['access_token']
         self.base_url = self.config_json['base_url']
         self.header = {'x-cachet-token': self.access_token}
-        self.components = {
-            'operational': [],
-            'performance_issues': [],
-            'partial_outage': [],
-            'major_outage': []
-        }
-        self.incidents = {
-            'investigating': [],
-            'identified': [],
-            'watching': [],
-            'fixed': []
-        }
-        self.passed_tests = []
-        self.failed_tests = []
-
-    def parseTestResults(self):
-        self.passed_tests = []
-        self.failed_tests = []
+        self.broken_components = []
+        self.open_incidents = []
 
-    def getAllStatuses(self):
+    def get_broken_components(self):
         response = requests.get(f'{self.base_url}/components', self.header)
-        print(response)
         for component in response.json()['data']:
             status = component['status']
-            if status == 1:
-                self.components['operational'].append(component)
-            elif status == 2:
-                self.components['performance_issues'].append(component)
-            elif status == 3:
-                self.components['partial_outage'].append(component)
-            elif status == 4:
-                self.components['major_outage'].append(component)
-
-    def getAllIncidents(self):
+            if status != 1:
+                self.broken_components.append(component)
+
+    def get_all_open_incidents(self):
         response = requests.get(f'{self.base_url}/incidents', self.header)
-        print(response)
         for incident in response.json()['data']:
             status = incident['status']
-            if status == 1:
-                self.incidents['investigating'].append(incident)
-            if status == 2:
-                self.incidents['identified'].append(incident)
-            if status == 3:
-                self.incidents['watching'].append(incident)
-            if status == 4:
-                self.incidents['fixed'].append(incident)
+            if status != 4:
+                self.open_incidents.append(incident)
+
+    def update_incident_status(self):
+        # map api version and name to Cachet component IDs
+        component_map = {
+            'v1/academic-disciplines': 3,
+            'v1/advisors': 4,
+            'v1/xeapps': 5,
+            'v1/beaverbus': 6,
+            'v1/directory': 7,
+            'v2/directory': 8,
+            'v1/finance': 9,
+            'v1/hr': 10,
+            'v1/identify': 11,
```

Comment:
We also have identities API v2. Thinking about it more, I think we should probably move this map to config file so that we can easily maintain it without modifying the code.